src/dependencies.jl

```diff
@@ -410,10 +418,20 @@ else
     end
 end
 
-lookup_soname(s) = lookup_soname(String(s))
-function lookup_soname(s::String)
-    have_sonames[] || read_sonames()
-    return get(sonames, s, "")
+if VERSION >= v"0.7.0-DEV.1287" # only use this where julia issue #22832 is fixed
+    lookup_soname(s) = lookup_soname(String(s))
+    function lookup_soname(s::String)
+        have_sonames[] || read_sonames()
+        return get(sonames, s, "")
+    end
+else
+    function lookup_soname(lib)
+        if is_linux() || (is_bsd() && !is_apple())
```

Comment:
Should be `Compat.Sys.islinux() || (Compat.Sys.isbsd() && !Compat.Sys.isapple())`, or stick a `using Compat.Sys: islinux, isapple, isbsd` at the top and drop the `Compat.Sys`s here.